sensor/common/networkflow/metrics/metrics.go

```diff
@@ -47,13 +49,24 @@ var (
 		Name:      "network_flow_misses_container_lookup",
 		Help:      "Total number of misses on container lookup for network flows",
 	})
-	ExternalFlowCounter = prometheus.NewCounter(prometheus.CounterOpts{
+	ExternalFlowCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Namespace: metrics.PrometheusNamespace,
 		Subsystem: metrics.SensorSubsystem.String(),
 		Name:      "network_flow_external_flows",
 		Help:      "Total number of external flows",
-	})
-
+	}, []string{"direction", "namespace"})
+	InternalFlowCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "network_flow_internal_flows",
+		Help:      "Total number of internal flows",
+	}, []string{"direction", "namespace"})
```

Comment:
Unfortunately, the descriptions of those two metrics are a bit tricky to formulate. I think that the current help strings are not accurate enough.

The metrics will count the numbers of external and internal flows every 30 seconds (this is how often the `enrichAndSend` can be called). On each execution, it may process flows that have already been processed in the previous execution in addition to the flows that are new. This would cause the metric to have a step-shape that would increase forever with jumps every 30s. While it does provide some more insights into  what sensor is doing, it may also lead to confusion for an untrained eye.

We may fix it in two ways:
- Make sure that the metric counts only unique flows (difficult)
- Adapt the help text (and optionally metric name) to better describe the situation (easy)

If I would be crafting the help text, I would start with something like: "Total number of internal|external flows (non-unique) processed by sensor in the enrichment"

Reply:
> Make sure that the metric counts only unique flows (difficult)

I thought this was handled by the `status.used` property, no? We mark this connection as used after we process it on the ticker, and the metric is only incremented if `!status.used `. Although I'm not sure if that's the correct understanding of this property.

Reply:
I am sure that `status.used` is not an ultimate protection against processing the same connection twice. It may happen that the same connection is seen differently based on the status of the deployments it involves (their IDs, for example).
I would need to go deeper to answer this with more precision. For the need of updating the help text of ma metric, this may be too much effort. However, if you feel that this investment is worth it, I am happy to dig deeper.

Reply:
Updated

Reply:
Simple and to the point - thanks for rewriting this!